app/service_objects/cocktail_counter.rb

```diff
@@ -0,0 +1,18 @@
+class CocktailCounter
+
+  # @param cocktail: Recipe
+  #
+  # Given a cocktail (or recipe), returns how many the user can make
+  # with their reagents.
+  def self.count_possible_cocktails(cocktail, user)
+    cocktail.matching_reagents(user).map do |amount, reagents|
+      next if amount.unitless?
+
+      total_amount = reagents.map(&:current_volume).reduce(&:+)&.convert_to(:ml) || Measured::Volume.new(0, :ml)
```

Comment:
This will combine different bottles that fulfill the same amount in a drink right?

Reply:
Yes it would, however, that might not be something we want here?